source/adapters/hip/event.cpp

```diff
@@ -58,7 +59,7 @@ ur_result_t ur_event_handle_t_::start() {
   ur_result_t Result = UR_RESULT_SUCCESS;
 
   try {
-    if (Queue->URFlags & UR_QUEUE_FLAG_PROFILING_ENABLE) {
+    if (Queue->URFlags & UR_QUEUE_FLAG_PROFILING_ENABLE || isTimestampEvent()) {
       // NOTE: This relies on the default stream to be unused.
       UR_CHECK_ERROR(hipEventRecord(EvQueued, 0));
```

Comment:
@JackAKirk - In https://github.com/intel/llvm/pull/12838 it seems like the submission time on HIP is giving weird values. I did a bit of digging and it seems to me like HIP is a little different from CUDA when checking timing-differences between events. Of particular interest here is the following line for [hipEventElapsedTime()](https://docs.amd.com/projects/HIP/en/docs-5.4.0/doxygen/html/group___event.html#gad4128b815cb475c8e13c7e66ff6250b7): 

> Events which are recorded in a NULL stream will block until all commands on all other streams complete execution, and then record the timestamp.

While what we expect here is to get an event with the current time, hence using an otherwise unused stream. Fixing it might be outside the scope of this PR, but a possible solution could be to lazily have a stream specifically for recording submission time of events, tied to the context. Similar could be used in the CUDA backend to avoid the assumption noted above.

Reply:
@konradkusiak97 this sounds like the same thing as your hip event timings issue?

Reply:
I have opened an issue about it here: https://github.com/intel/llvm/issues/12904.